### PR TITLE
More circuit entity outputs

### PR DESCRIPTION
**Pull request by chloecat0**
## About The Pull Request
Adds a scanner shell which outputs an atom when used on something. Adds a user and triggered port to airlock shells which output when the airlock is touched. Adds a payer port to the moneybot shell.

## Why It's Good For The Game
Right now the only way to output an atom is to use a voice activator circuit, a circuit multitool or an mmi which is quite limiting, this PR adds more so that circuits can be more versatile.

## Changelog
:cl:
add: Added scanner circuit shell
tweak: Added a user and triggered port to airlock shells
tweak: Added a payer port to moneybot shells
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
Please list prs you ported and credit original authors.

**Comment by chloecat0** [comment]
Nothing was actually ported the wiremod things are just called ports

**Comment by godspeedfunnyman** [comment]
this is neat! finally i'll be able to do the drone that's able to spam pull everyone that are near them

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -416,6 +416,8 @@
 				if(G.siemens_coefficient)//not insulated
 					new /datum/hallucination/shock(H)
 					return
+	if(SEND_SIGNAL(src, COMSIG_AIRLOCK_TOUCHED, user) & COMPONENT_PREVENT_OPEN)
+		return
```
Comment (Archanial) [inline code/game/machinery/doors/airlock.dm:420]: This won't send anything if the target got zapped, is this intentional?

File: code/modules/wiremod/shell/scanner.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * # Scanner
+ *
+ * A handheld device which scans things.
+ */
+/obj/item/scanner
+	name = "scanner"
+	icon = 'icons/obj/assemblies/electronic_setups.dmi'
+	icon_state = "setup_small_hook"
+	item_state = "electronic"
+	//worn_icon_state = "electronic"		//remember to change it later lol
```
Comment (Archanial) [inline code/modules/wiremod/shell/scanner.dm]: :flushed:

File: code/modules/wiremod/shell/airlock.dm [hunk 3]
```diff
@@ -110,6 +121,12 @@
 	is_open.set_output(FALSE)
 	closed.set_output(COMPONENT_SIGNAL)
 
+/obj/item/circuit_component/airlock/proc/on_airlock_touched(datum/source, user)
+	SIGNAL_HANDLER
+	. = COMPONENT_PREVENT_OPEN
```
Comment (Archanial) [inline code/modules/wiremod/shell/airlock.dm:126]: okay but won't it always prevent airlock from opening?

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (chloecat0) [reply code/game/machinery/doors/airlock.dm:420]: Yeah that way it works similarly to normal airlocks

File: code/modules/wiremod/shell/airlock.dm (hunk 3 above)
Comment (chloecat0) [reply code/modules/wiremod/shell/airlock.dm:126]: There's no way to open the airlock without using the circuit anyway so it doesn't really change anything

File: code/modules/wiremod/shell/scanner.dm (hunk 2 above)
Comment (chloecat0) [reply code/modules/wiremod/shell/scanner.dm]: the compact remote and controller shells have this too so I just left it

Comment (Archanial) [reply code/modules/wiremod/shell/scanner.dm]: delete it, we're not using this

Comment (chloecat0) [reply code/modules/wiremod/shell/scanner.dm]: sure

**Comment by Sinestia** [comment]
did you PR this upstream too?

you probably should

**Review by Archanial — APPROVED** [review]
Good stuff

**Comment by utilbird** [comment]
> did you PR this upstream too?
you probably should

Spy

**Review by utilbird — APPROVED** [review]

**Pull request merged**